src/agentlab/analyze/agent_xray.py

```diff
@@ -557,6 +593,30 @@ def update_agent_info():
         return None
 
 
+def submit_action(input_text):
+    global info
+    agent_info = info.exp_result.steps_info[info.step].agent_info
+    chat_messages = deepcopy(agent_info.get("chat_messages", ["No Chat Messages"])[:2])
+    assert isinstance(chat_messages[1], BaseMessage), "Messages should be langchain messages"
+
+    chat = ChatOpenAI(name="gpt-4o-mini")
+    chat_messages[1].content = input_text
```

Comment:
In the long run, we may want to modify the system message or other messages as well. We'll see how we can play with the UI. But for now it's good :)